### PR TITLE
documentation + ip

### PR DESCRIPTION
**Pull request by greut**
Signed-off-by: Yoan Blanc <yoan.blanc@exoscale.ch>

### CODE REVIEW

**Comment by greut** [comment]
I've added a little list of what does what regarding to the CS api. Not complete yet...

https://github.com/exoscale/egoscale/wiki/api

File: ip.go [hunk 1]
```diff
@@ -28,57 +29,109 @@ func (exo *Client) AssociateIpAddress(profile IpAddressProfile, async AsyncInfo)
...
-	return r.Success, nil
+	if !r.Success {
+		return fmt.Errorf("Cannot disassociateIpAddress. %s", r.DisplayText)
+	}
+
+	return nil
+}
+
+// GetAllIpAddresses returns all the public IP addresses
+func (exo *Client) GetAllIpAddresses() ([]*IpAddress, error) {
```
Comment (marcaurele) [inline ip.go]: Will be useful to have another shortcut to list all IPs in a zone.

File: ip.go [hunk 2]
```diff
@@ -28,57 +29,109 @@ func (exo *Client) AssociateIpAddress(profile IpAddressProfile, async AsyncInfo)
...
+
+	var r ListPublicIpAddressesResponse
+	if err := json.Unmarshal(resp, &r); err != nil {
+		return nil, err
+	}
+
+	return r.PublicIpAddress, nil
+}
+
+// TODO move to nic.go
```
Comment (marcaurele) [inline ip.go]: Not doing it now?

File: ip.go [hunk 3]
```diff
@@ -28,57 +29,109 @@ func (exo *Client) AssociateIpAddress(profile IpAddressProfile, async AsyncInfo)
...
+}
+
+// GetAllIpAddresses returns all the public IP addresses
+func (exo *Client) GetAllIpAddresses() ([]*IpAddress, error) {
+	params := url.Values{}
+
+	return exo.ListPublicIpAddresses(params)
+}
+
+func (exo *Client) ListPublicIpAddresses(params url.Values) ([]*IpAddress, error) {
```
Comment (marcaurele) [inline ip.go]: Documentation line is missing on this one.

File: ip.go (hunk 2 above)
Comment (greut) [reply ip.go]: gotcha!

File: ip.go (hunk 1 above)
Comment (greut) [reply ip.go]: yagni!

File: vm.go [hunk 4]
```diff
@@ -86,10 +88,11 @@ func (exo *Client) doVirtualMachine(action string, params url.Values, async Asyn
 	return r.VirtualMachine, nil
 }
 
-func (exo *Client) GetVirtualMachine(id string) (*VirtualMachine, error) {
+// GetVirtualMachine
+func (exo *Client) GetVirtualMachine(virtualMachineId string) (*VirtualMachine, error) {
```
Comment (marcaurele) [inline vm.go]: A `ListVirtualMachine` by name would be useful I think too

Comment (greut) [reply vm.go]: eventually, indeed.

**Pull request merged**